### PR TITLE
VUMIGO-250 adding option to account for SMS confirmations.

### PR DESCRIPTION
**Pull request by smn**
These are largely commits pulled out of #126 to make the reviews easier.
It adds a boolean for whether or not SMS confirmations are wanted.
If that's checked then a valid phone number is required.

Also adds some updates to the account form, requiring the old / original
password be supplied before any changes are accepted.

Also using BootstrapForm better.

### CODE REVIEW

File: go/account/forms.py [hunk 1]
```diff
@@ -1,21 +1,83 @@
...
+        Make a best effort guess at determining whether this msisdn
+        is a valid msisdn
+        """
+        msisdn = self.cleaned_data['msisdn'].lstrip('+')
+        if not msisdn:
+            return ''
+
+        if not (len(msisdn) > 5 and
+                all(c.isdigit() for c in msisdn)):
+            raise forms.ValidationError('Please provide a valid phone number.')
```
Comment (jerith) [inline go/account/forms.py:70]: This also disallows spaces, dashes, parens, etc. which are often handy to use when typing phone numbers. Maybe strip these out at the same time as the leading `+`?

File: go/account/tests.py [hunk 2]
```diff
@@ -1,16 +1,164 @@
...
-        """
-        self.assertEqual(1 + 1, 2)
+    def setUp(self):
+        super(AccountTestCase, self).setUp()
+        self.setup_riak_fixtures()
+        self.client = Client()
+        self.client.login(username='username', password='password')
+
+    def tearDown(self):
+        pass
```
Comment (jerith) [inline go/account/tests.py]: Are we explicitly overriding a superclass `tearDown()` here? If so, can we document this and explain why? If not, can we remove this?

**Comment by smn** [comment]
@jerith I've addressed your review comments, can I get a re-review?

**Comment by jerith** [comment]
It would probably make more sense to keep these new fields on the Account object in Riak. The Django user stuff shouldn't be much more than auth information.

**Comment by jerith** [comment]
:+1:

**Pull request merged**